### PR TITLE
IoC: Optionally use objensis for proxy creation

### PR DESCRIPTION
**Pull request by selckin**
This allows creating proxies for concrete classes without a default
constructor.

Guice IoC cache proxies similar to spring

-----

It may need some  more finishing touches, but would something like this be considered for inclusion?

### CODE REVIEW

**Comment by martin-g** [comment]
Hi,

I see this PR introduces some API changes.
We want to release Wicket 7.0.0 next week so I am a bit hesitant to merge it.
Please start a discussion at dev@wicket.apache.org. If others think it is something important then we will merge it.

**Comment by selckin** [comment]
Now with less api changes

File: wicket-guice/src/main/java/org/apache/wicket/guice/GuiceFieldValueFactory.java [hunk 1]
```diff
@@ -64,15 +68,34 @@ public Object getFieldValue(final Field field, final Object fieldOwner)
 				{
 					boolean optional = injectAnnotation != null && injectAnnotation.optional();
 					Annotation bindingAnnotation = findBindingAnnotation(field.getAnnotations());
-					final IProxyTargetLocator locator = new GuiceProxyTargetLocator(field, bindingAnnotation, optional);
+					final GuiceProxyTargetLocator locator = new GuiceProxyTargetLocator(field, bindingAnnotation, optional);
+
+					Object cachedValue = cache.get(locator);
```
Comment (martin-g) [inline wicket-guice/src/main/java/org/apache/wicket/guice/GuiceFieldValueFactory.java]: The caching is not related to Objenesis, right?

Comment (selckin) [inline wicket-guice/src/main/java/org/apache/wicket/guice/GuiceFieldValueFactory.java]: It's not, but i'm not sure about the objensis performance, and the spring ioc also caches the proxies, so it keeps those 2 closer together aswel

Comment (martin-g) [inline wicket-guice/src/main/java/org/apache/wicket/guice/GuiceFieldValueFactory.java]: I'm going to revert all non-Objenesis caches.
This should be in separate ticket.

**Pull request merged**